sw/device/lib/testing/rstmgr_testutils.c

```diff
@@ -83,7 +73,11 @@ void rstmgr_testutils_post_reset(
         "Unexpected reset_info CSR mismatch, expected 0x%x, got 0x%x",
         expected_reset_info, actual_reset_info);
 
-  rstmgr_testutils_compare_alert_info(rstmgr, expected_alert_dump,
-                                      alert_dump_size);
-  rstmgr_testutils_compare_cpu_info(rstmgr, expected_cpu_dump, cpu_dump_size);
+  if (expected_alert_dump != NULL && alert_dump_size != 0) {
```

Comment:
If the caller wants to avoid the checks it can pass a null or zero sized buffer, keeping the compare code clean.